module/applications/sheets-configs/prototype-token-config.mjs

```diff
@@ -34,8 +40,89 @@ export default class DhPrototypeTokenConfig extends foundry.applications.sheets.
 
         return context;
     }
+
+    _attachPartListeners(partId, htmlElement, options) {
+        super._attachPartListeners(partId, htmlElement, options);
+
+        switch (partId) {
+            case 'appearance':
+                htmlElement
+                    .querySelector('#dhTokenSize')
+                    ?.addEventListener('change', this.onTokenSizeChange.bind(this));
+                break;
+        }
+    }
+
+    /** @inheritDoc */
+    _previewChanges(changes) {
+        if (!changes || !this._preview) return;
+
+        const tokenSizeSelect = this.element?.querySelector('#dhTokenSize');
+        if (this.token.actor && tokenSizeSelect && tokenSizeSelect.value !== 'custom') {
+            const tokenSizes = game.settings.get(CONFIG.DH.id, CONFIG.DH.SETTINGS.gameSettings.Homebrew).tokenSizes;
+            const tokenSize = tokenSizes[tokenSizeSelect.value];
+            changes.width = tokenSize;
+            changes.height = tokenSize;
+        }
+
+        const deletions = { '-=actorId': null, '-=actorLink': null };
+        const mergeOptions = { inplace: false, performDeletions: true };
+        this._preview.updateSource(mergeObject(changes, deletions, mergeOptions));
+
+        if (this._preview?.object?.destroyed === false) {
+            this._preview.object.initializeSources();
+            this._preview.object.renderFlags.set({ refresh: true });
+        }
+    }
+
+    async onTokenSizeChange(event) {
+        const value = event.target.value;
+        const tokenSizeDimensions = this.element.querySelector('#tokenSizeDimensions');
+        if (tokenSizeDimensions) {
+            const disabled = value !== 'custom';
+
+            tokenSizeDimensions.dataset.tooltip = disabled
+                ? game.i18n.localize('DAGGERHEART.APPLICATIONS.TokenConfig.actorSizeUsed')
+                : '';
+
+            const disabledIcon = tokenSizeDimensions.querySelector('i');
+            if (disabledIcon) {
+                disabledIcon.style.opacity = disabled ? '' : '0';
+            }
+
+            const dimensionsInputs = tokenSizeDimensions.querySelectorAll('.form-fields input');
+            for (const input of dimensionsInputs) {
+                input.disabled = disabled;
+            }
+        }
+    }
+
+    /**
+     * Process form submission for the sheet
+     * @this {PrototypeTokenConfig}
+     * @type {ApplicationFormSubmission}
+     */
+    static async #onSubmit(event, form, formData) {
+        const submitData = this._processFormData(event, form, formData);
+        submitData.detectionModes ??= []; // Clear detection modes array
+        this._processChanges(submitData);
+        const changes = { prototypeToken: submitData };
+
+        const tokenSizeSelect = this.element.querySelector('#dhTokenSize');
+        if (tokenSizeSelect && this.actor) {
+            if (tokenSizeSelect.value !== this.actor.system.size) {
```

Comment:
I think for this it would be neater to be one if. If it formats to two lines you could store isSizeDifferent in a variable.

Reply:
Shortened it to one if :+1: